### ISSUE TITLE
Plotting fails only for specific range

### ISSUE THREAD

**Issue by Ceronsand** [MakieOrg/Makie.jl#931]
Steps to reproduce:
```julia
]activate --temp
add GLMakie
using GLMakie
plot([1.4030269373737038e6, 1.4030269377481483e6, 1.4030269381214702e6])
```
ERROR: ArgumentError: range step cannot be zero

Unfortunately i could not yet narrow it down to a reproducible range, error dissapears if something greater than the 6th decimal is changed . Changing the values slightly allows plotting, but the axis doesn't seem aligned properly anymore.
For example:
```julia
plot([1.4030269373737038e6, 1.4030269377481483e6, 1.4030279381214702e6])
```

System Configuration:
Win10, Julia 1.6.1, GLMakie@0.2.8

**Comment by jkrumbiegel** [comment]
The problem is that Makie uses Float32 a lot internally, as this is what GPUs want. And your values are very close together, which means some algorithm that tries to find steps between your limits finds zero-width range steps. I have just tried this out, out of interest:

```julia
julia> nfloats(a, b) = begin
           n = 0
           while true
               a = nextfloat(a)
               n += 1
               a == b && break
           end
           n
       end

julia> a = Float32(1.4030269373737038e6)
1.4030269f6

julia> b = Float32(1.4030269381214702e6)
1.403027f6

julia> nfloats(a, b)
1
```

So there is just one step from your low to your high value in Float32. There's currently no better solution than to scale your values up in Float64 before plotting.

**Comment by humphreylee** [comment]
Encountered this issue. The workaround works. Suggest better approach to handle this. Thanks.

**Comment by musoke** [comment]
I have also run into this error.  

```
using Pkg
Pkg.status("GLMakie")

using GLMakie
using Statistics

b = 0:1e-7:2e-6

lines(b)  # works

lines(b .+ 50)  #error
```

A better solution would be good; it's not as if plotting values with a large mean and small variance is an obscure task.

**Comment by asinghvi17** [comment]
Aside from the type requirements (e.g. Rect2f, Point2f, Vec2f), is there anything in Makie or MakieCore (not the backends) which relies on values being Float32? 

 If not, I could take a look at making Makie independent of float type, possibly accepting anything which is <: AbstractFloat or Real, then converting them in e.g. `gl_convert` for GLMakie, and leaving them unconverted for CairoMakie.

**Comment by musoke** [comment]
@ffreyer, I know nearly nothing about the internals of makie, but that sounds like a sensible solution.

**Comment by islent** [comment]
Truncating the array to `Float16` saves the world:
```julia
julia> a = [31.999999051331837, 31.99999905020038, 31.99999904907261, 31.999999047948524, 31.99999904682809, 31.99999904571132, 31.999999044598198, 31.999999043488696, 31.999999042382818, 31.999999041280553, 31.99999904018189]
11-element Vector{Float64}:
 31.999999051331837
 31.99999905020038
 31.99999904907261
 31.999999047948524
 31.99999904682809
 31.99999904571132
 31.999999044598198
 31.999999043488696
 31.999999042382818
 31.999999041280553
 31.99999904018189

julia> Makie.lines(a)
Error showing value of type Makie.FigureAxisPlot:
ERROR: ArgumentError: range step cannot be zero
...

julia> Makie.lines(Float32.(a))
Error showing value of type Makie.FigureAxisPlot:
ERROR: ArgumentError: range step cannot be zero
...

julia> Makie.lines(Float16.(a))
#! It works!
```

Related issues:
- https://github.com/JuliaPlots/Makie.jl/issues/1040
- https://github.com/JuliaPlots/Makie.jl/issues/1579

**Comment by musoke** [comment]
Both @Ceronsand's [MWE](https://github.com/MakieOrg/Makie.jl/issues/931#issue-870888541) and [my MWE](https://github.com/MakieOrg/Makie.jl/issues/931#issuecomment-1147743448) are fixed with GLMakie v0.10.0.
